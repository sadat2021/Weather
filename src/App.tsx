import React from "react";
import Main from "./page";
import "./styles.scss";

function App() {
  return (
    <div className="App">
      <Main />
    </div>
  );
}

export default App;
