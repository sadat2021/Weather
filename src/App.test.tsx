import React from "react";
import { render, screen } from "@testing-library/react";
import App from "./App";

describe("renders application", () => {
  it("renders form and card", () => {
    const { container } = render(<App />);
    expect(screen.getByText("Title")).toBeInTheDocument();
    expect(screen.getByText("Temperature")).toBeInTheDocument();
    expect(screen.getAllByText("Wind")).toHaveLength(2);
    expect(screen.getByText("TITLE OF WIDGET")).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
