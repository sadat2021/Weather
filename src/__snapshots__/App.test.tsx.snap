// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders application renders form and card 1`] = `
<div>
  <div
    class="App"
  >
    <div
      class="main"
    >
      <div
        class="form"
      >
        <div>
          <label
            class="label"
            for="title"
          >
            Title 
            <small>
              15 characters
            </small>
          </label>
          <input
            class="text-input"
            data-testid="title-input"
            id="title"
            maxlength="15"
            placeholder="Title of widget"
            type="text"
            value=""
          />
        </div>
        <div>
          <span
            class="label"
          >
            Temperature
          </span>
          <div>
            <input
              checked=""
              data-testid="c-input"
              id="deg-c"
              name="temperature"
              type="radio"
            />
            <label
              for="deg-c"
            >
              <sup>
                °
              </sup>
              C
            </label>
          </div>
          <div>
            <input
              data-testid="f-input"
              id="deg-f"
              name="temperature"
              type="radio"
            />
            <label
              for="deg-f"
            >
              <sup>
                °
              </sup>
              F
            </label>
          </div>
        </div>
        <div>
          <span
            class="label"
          >
            Wind
          </span>
          <div>
            <input
              checked=""
              data-testid="wind-on"
              id="on"
              name="wind"
              type="radio"
            />
            <label
              for="on"
            >
              On
            </label>
          </div>
          <div>
            <input
              data-testid="wind-off"
              id="off"
              name="wind"
              type="radio"
            />
            <label
              for="off"
            >
              Off
            </label>
          </div>
        </div>
      </div>
      <div
        class="divider"
      />
      <div
        class="card"
      >
        <div>
          <h3
            data-testid="card-title"
          >
            TITLE OF WIDGET
          </h3>
          <div
            class="display-details"
          >
            <img
              alt="icon"
              class="img"
              data-testid="icon"
              src="sunny-cloudy.jpeg"
            />
            <div
              class="details"
            >
              <span
                data-testid="name"
              />
              <span
                data-testid="temperature"
              >
                --
                <span>
                  °
                </span>
              </span>
              <span
                data-testid="wind"
              >
                <span
                  data-testid="wind-display"
                >
                  Wind 
                </span>
                 
                <span
                  data-testid="wind-deg"
                >
                  --
                   
                </span>
                 
                <span
                  data-testid="wind-speed"
                >
                  --
                  km/h
                </span>
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
